File: fru1tworld/extra/leetcode75/Graph-DFS/leet841/leet841.cpp
class Solution {
public:
    bool visited[1005]; 
    void dfs(vector<vector<int>>& rooms, int now){
        visited[now] = true; 
        for(int i=0; i<rooms[now].size(); i++){
            int next = rooms[now][i];
            if(!visited[next]){
                dfs(rooms, next);
            }
        }
    }
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        dfs(rooms, 0);
        for(int i=0; i<rooms.size(); i++){
            if(!visited[i]) return false; 
        }
        return true; 
    }
};

File: iam454/week24/leet841.js
/**
 * @param {number[][]} rooms
 * @return {boolean}
 */
var canVisitAllRooms = function (rooms) {
  const q = [];
  const N = rooms.length;
  const visited = new Array(N).fill(0);
  q.push([0]);

  while (q.length) {
    const keys = q.shift();

    for (let i = 0; i < keys.length; i++) {
      if (visited[keys[i]] === 0) {
        visited[keys[i]] = 1;
        q.push(rooms[keys[i]]);
      }
    }
  }

  return visited.every((item) => item === 1);
};
